src/main/java/calculator/Parser.java

```diff
@@ -0,0 +1,76 @@
+package calculator;
+
+import java.util.Arrays;
+import java.util.InputMismatchException;
+import java.util.List;
+import java.util.Objects;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+public final class Parser {
+    private static final Pattern FORBIDDEN_DELIMITER_PATTERN = Pattern.compile("\\.|\\?|\\+|\\*|\\^|\\$|\\|");
+    private static final Pattern CUSTOM_DELIMITER_PATTERN = Pattern.compile("//(.)\n(.*)");
+    private static final String CUSTOM_DELIMITER_PREFIX = "//";
+    private static final String DEFAULT_DELIMITER = ",|:";
+    private static final int DELIMITER_CAPTURE_POSITION = 1;
+    private static final int TARGET_CAPTURE_POSITION = 2;
+    private final String input;
```

Comment:
상수와 필드는 나누어 주는 것이 좋습니다 :)
```suggestion
    private static final int TARGET_CAPTURE_POSITION = 2;
    
    private final String input;
```